portality/view/publisher.py

```diff
@@ -222,9 +224,12 @@ def metadata():
             if not enough_authors:
                 return render_template('publisher/metadata.html', form=form, author_error=True)
             else:
-                xwalk = article.FormXWalk()
+                # xwalk = article.FormXWalk()
+                xwalk = ArticleFormXWalk()
                 art = xwalk.crosswalk_form(form)
-                art.save()
+                articleService = DOAJ.articleService()
+                articleService.create_article(art, current_user._get_current_object())
+                # art.save()
```

Comment:
articleService.create_article does indeed save the article if it's successful, these commented out lines should be removed.

Reply:
done